### PR TITLE
Implemented according to discussions.

### PR DESCRIPTION
**Pull request by afinne**
Unit tests are not fully implemented due to differences in Sqlite and Postgres syntax.

We might need to revisit the logic for "plaque", I'm starting to lean towards actually deleting the row for plaque to make it possible to difference between "mark as deleted" and "delete".
We might also implement support for an `_archived` field to support cleaning of old/finished (deleted in couchdb), but not "purged"

### CODE REVIEW

File: lib/couchdb_to_sql/table_deleted_marker.rb
Comment (perlun) [inline lib/couchdb_to_sql/table_deleted_marker.rb]: `unless/else` is generally seen as a bad pattern, because it's unnecessarily hard to follow. Please either go with `if !foo` or `if foo` and re-arrange the conditional blocks.

File: test/unit/table_deleted_marker_test.rb
Comment (perlun) [inline test/unit/table_deleted_marker_test.rb:67]: Thanks for _trying_ to add tests here! We'll have to complete it some other time if we are unable to find a good way to complete it now.

**Review by perlun — CHANGES_REQUESTED** [review]
Some cleanups that should happen before merge, but on the general whole, a very fine change - thanks for the effort. 👍

File: test/unit/table_deleted_marker_test.rb
Comment (afinne) [reply test/unit/table_deleted_marker_test.rb:67]: Just for the record here why Mr @perlun said _trying_. The tests are running on Sqlite, while we run our real code on Postgres. The syntax for upserts (insert_conflict) is different for Sqlite and Postgres.

Comment (perlun) [reply test/unit/table_deleted_marker_test.rb:67]: Indeed. So the only way to properly test it right now would be to use a real Postgres database for testing. It's _doable_, but it mean we would have to add Rake tasks for setting it up, run it on Travis, etc etc.

File: lib/couchdb_to_sql/changes.rb
Comment (perlun) [inline lib/couchdb_to_sql/changes.rb:140]: I'm fine with the change, but I'm also fine with the original way this was written. Rubocop is wrong in this regard. 😛

File: lib/couchdb_to_sql/document_handler.rb
Comment (perlun) [inline lib/couchdb_to_sql/document_handler.rb:32]: Not related to this change, but: `TableDestroyer` is really, _really_ a bad name of the class here. The casual reader will understand this as "destroy the given table", which is not at all what happens...

(I was probably the one who came up with that silly name to begin with.)

**Review by perlun — APPROVED** [review]
Looks good now, thanks. 👍 Bonus points for adding tests for the new semantics.

I also made notes on things which are more of "general improvement, can be done someday" nature. They do not block merging this PR.

**Pull request merged**
